### PR TITLE
Make InstructPix2Pix Training Script torch.compile compatible 

### PR DESCRIPTION
**Pull request by charchit7**
# What does this PR do?

Part of #6503 for `instruct pix2pix`

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@sayakpaul 🔢

### CODE REVIEW

File: examples/instruct_pix2pix/train_instruct_pix2pix.py [hunk 1]
```diff
@@ -965,14 +971,14 @@ def collate_fn(examples):
     # Create the pipeline using the trained modules and save it.
     accelerator.wait_for_everyone()
     if accelerator.is_main_process:
-        unet = accelerator.unwrap_model(unet)
+        unet = unet
```
Comment (sayakpaul) [inline examples/instruct_pix2pix/train_instruct_pix2pix.py:974]: What's this? :D

File: examples/instruct_pix2pix/train_instruct_pix2pix.py [hunk 2]
```diff
@@ -965,14 +971,14 @@ def collate_fn(examples):
...
         if args.use_ema:
             ema_unet.copy_to(unet.parameters())
 
         pipeline = StableDiffusionInstructPix2PixPipeline.from_pretrained(
             args.pretrained_model_name_or_path,
-            text_encoder=accelerator.unwrap_model(text_encoder),
-            vae=accelerator.unwrap_model(vae),
```
Comment (sayakpaul) [inline examples/instruct_pix2pix/train_instruct_pix2pix.py:975]: Why?

**Review by sayakpaul — COMMENTED** [review]
Thanks!

Left a couple of comments.

File: examples/instruct_pix2pix/train_instruct_pix2pix.py (hunk 1 above)
Comment (charchit7) [reply examples/instruct_pix2pix/train_instruct_pix2pix.py:974]: Ahhh, changing.

File: examples/instruct_pix2pix/train_instruct_pix2pix.py (hunk 2 above)
Comment (charchit7) [reply examples/instruct_pix2pix/train_instruct_pix2pix.py:975]: I was testing output forgot the change. :D

**Comment by charchit7** [comment]
> Thanks!
> 
> Left a couple of comments.

Thank you, fixed.

**Review by sayakpaul — APPROVED** [review]
Thanks!

**Pull request merged**